### ISSUE TITLE
[MJAR-279] maven-jar-plugin after upgrade of Maven to 3.8.3 doesn't include classpath to MANIFEST 

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-jar-plugin#426]
**[Jakub Moravec](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jakubmoravec)** opened **[MJAR-279](https://issues.apache.org/jira/browse/MJAR-279?redirect=false)** and commented

```
Plugin configuration:
```

```

<plugin>
  <groupId>org.apache.maven.plugins</groupId>
  <artifactId>maven-jar-plugin</artifactId>
  <configuration>
    <archive>
      <index>false</index>
      <manifest>
        <addClasspath>true</addClasspath>
        <useUniqueVersions>false</useUniqueVersions>
        <mainClass>eu.profinit.manta.platform.cli.Main</mainClass>
        <addDefaultImplementationEntries>true</addDefaultImplementationEntries>
        <addDefaultSpecificationEntries>true</addDefaultSpecificationEntries>
      </manifest>
    </archive>
  </configuration>
</plugin>
```

Diff of MANIFEST.MF before (`mvn clean install` executed with Maven 3.5.4) and after (`mvn clean install` executed with Maven 3.8.3) is attached. 

 

 

---

**Affects:** 3.2.0

**Attachments:**
- [debug_3_5_4_OK.log](https://issues.apache.org/jira/secure/attachment/13035335/debug_3_5_4_OK.log) (_821.71 kB_)
- [debug_3_6_3_OK.log](https://issues.apache.org/jira/secure/attachment/13035336/debug_3_6_3_OK.log) (_821.14 kB_)
- [debug_3_8_1_OK.log](https://issues.apache.org/jira/secure/attachment/13035337/debug_3_8_1_OK.log) (_821.26 kB_)
- [debug_3_8_2_OK.log](https://issues.apache.org/jira/secure/attachment/13035338/debug_3_8_2_OK.log) (_816.10 kB_)
- [debug_3_8_3_NOK.log](https://issues.apache.org/jira/secure/attachment/13035339/debug_3_8_3_NOK.log) (_810.17 kB_)
- [image-20211025-122928.png](https://issues.apache.org/jira/secure/attachment/13035313/image-20211025-122928.png) (_35.86 kB_)

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Please provide `-X` output.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Please try 3.6.3, 3.8.1, 3.8.2 as well.

**Comment by jira-importer** [comment]
**[Jakub Moravec](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jakubmoravec)** commented

Hi [Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o), thanks for looking into this. Please find the debug logs attached. 3.8.3 is the only version that doesn't work.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Thanks, I will look into it.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

This is, though, somewhat confusing since the plugin does not change. Maybe a regression between 3.8.2 and 3.8.3. Are you maybe able to build Maven from master-3.8.x branch and retry? There is a current revert on that branch which Amy already solve your issue.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

The logfile says you are still on: `--- maven-jar-plugin:2.4:jar (default-jar) @ manta-platform-cli ---`
Can you upgrade to 3.2.0 and see whether there is a change?

**Comment by jira-importer** [comment]
**[Greg A](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=grega)** commented

Same issue experienced here. Happens only for maven 3.8.3 when jar plugin version is 2.4

Fixed by upgrading to jar plugin 3.2.0 or 3.1.2

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

`Last Published: 2012-01-26  | Version: 2.4`. Not going to investigate this old version. [Jakub Moravec](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jakubmoravec), waiting for your upgrade tests.

**Comment by jira-importer** [comment]
**[Jakub Moravec](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jakubmoravec)** commented

[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o) I can confirm it works with both versions. Thank you very much for your help!
